### PR TITLE
LUCENE-9514: Include TermVectorsWriter in DWPT accounting

### PR DESCRIPTION
**Pull request by s1monw**
TermVectorsWriter might consume some heap space memory that
can have a significant impact on decisions made in the IW if
writers should be stalled or DWPTs should be flushed if memory
settings are small in IWC and flushes are frequent. This change adds
RAM accounting to the TermVectorsWriter since it's part of the
DWPT lifecycle and not just present during flush.

### CODE REVIEW

File: lucene/codecs/src/java/org/apache/lucene/codecs/simpletext/SimpleTextTermVectorsWriter.java [hunk 1]
```diff
@@ -193,4 +189,9 @@ private void write(BytesRef bytes) throws IOException {
   private void newLine() throws IOException {
     SimpleTextUtil.writeNewline(out);
   }
+
+  @Override
+  public long ramBytesUsed() {
+    return scratch.get().bytes.length;
```
Comment (mikemccand) [inline lucene/codecs/src/java/org/apache/lucene/codecs/simpletext/SimpleTextTermVectorsWriter.java:195]: Ha.

File: lucene/core/src/java/org/apache/lucene/index/StoredFieldsConsumer.java [hunk 2]
```diff
@@ -42,6 +44,7 @@
   protected void initStoredFieldsWriter() throws IOException {
     if (writer == null) { // TODO can we allocate this in the ctor? we call start document for every doc anyway
       this.writer = codec.storedFieldsFormat().fieldsWriter(directory, info, IOContext.DEFAULT);
+      accountable = writer;
```
Comment (mikemccand) [inline lucene/core/src/java/org/apache/lucene/index/StoredFieldsConsumer.java:47]: Hmm what is happening here?  Are we using this new `accountable` member somewhere?  I don't see it in the diffs?  Oh, I see, it is returned in `DefaultIndexingChain.getChildResources()`, ok!  Maybe add comment above its declaration explaining that our parent/owning `DefaultIndexingChain` returns/uses it?

File: lucene/core/src/java/org/apache/lucene/util/Accountable.java [hunk 3]
```diff
@@ -41,4 +41,8 @@
     return Collections.emptyList();
   }
 
+  /**
+   * An accountable that always returns 0
+   */
+  Accountable NULL_ACCOUNTABLE = () -> 0;
```
Comment (mikemccand) [inline lucene/core/src/java/org/apache/lucene/util/Accountable.java:47]: Interface members are always `static final` right?

**Review by mikemccand — APPROVED** [review]
Thanks @s1monw.  Surprising `TermVectorsWriter` might use non-trivial RAM!

File: lucene/core/src/java/org/apache/lucene/util/Accountable.java (hunk 3 above)
Comment (s1monw) [reply lucene/core/src/java/org/apache/lucene/util/Accountable.java:47]: yes

Comment (uschindler) [reply lucene/core/src/java/org/apache/lucene/util/Accountable.java:47]: To avoid such questions, you can be explicit.

Comment (s1monw) [reply lucene/core/src/java/org/apache/lucene/util/Accountable.java:47]: sure, but do we then initialize all booleans with `= false`? I thinks it's totally obsolet.

**Pull request merged**